thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx

```diff
@@ -32,107 +34,125 @@ const StyledGrid = withStyles((theme) => ({
 export const AlertTemplateSectionedProperties: FunctionComponent<AlertTemplateSectionedPropertiesProps> =
     ({ groupedProperties, handlePropertyValueChange }) => {
         const { t } = useTranslation();
+        const ordeeredGroupedProperties: Record<
+            string,
+            Record<string, PropertyRenderConfig[]>
+        > = {};
+        /* The idea is to display properties in a specifc order. We have that order pre-defined.
+            In case there is a property that belongs to a step that is not in pre-defined order,
+            we push it at the second last place, just before the properties of "OTHER" step
+        */
+        const keysPresent = Object.keys(groupedProperties);
+        ALERT_PROPERTIES_DEFAULT_ORDER.forEach((alertProperty) => {
+            if (alertProperty !== ALERT_PROPERTIES_DEFAULT_STEP.STEP) {
+                if (groupedProperties[alertProperty]) {
+                    ordeeredGroupedProperties[alertProperty] =
+                        groupedProperties[alertProperty];
+                }
+            }
+        });
+        const keysNotInDefaultOrder = keysPresent.filter(
+            (key) => !ALERT_PROPERTIES_DEFAULT_ORDER.includes(key)
+        );
+        if (!isEmpty(keysNotInDefaultOrder)) {
+            keysNotInDefaultOrder.forEach((key) => {
```

Comment:
are we just doing this in component directly?
wont it cause a lot of re-renders?

imo we can move this to a util function and save the value in useMemo / useState

Reply:
It wont cause any re-renders as we are creating a new object in the form we want from the incoming prop. there is no state update happening in between. Once we have the object ready we iterate over it render the component. Moving to util would be ideal if there is a repetitive use case for this kind of manipulation which we don't have as this very specific to this flow. If in future needs arises where we want to do something similar we can think and move it to some util